include/boost/beast/core/impl/file_posix.ipp

```diff
@@ -284,6 +284,7 @@ read(void* buffer, std::size_t n, error_code& ec) const
         if(result == 0)
         {
             // short read
+            ec = net::error::eof;
```

Comment:
I don't think we should return a `net::error` from any "...file" classes. These are supposed to be independent of networking, in fact at some point it should be factored out into its own library because it is generally useful. If you think about it, returning this error could confuse an HTTP read with a file body into thinking the peer shut down the socket, when it didn't.